.github/workflows/release.yaml

```diff
@@ -0,0 +1,912 @@
+name: Build and release
+
+on:
+  push:
+    tags:
+      - '*'
+  schedule:
+    - cron: '0 2 * * *'
+env:
+  CABAL_CACHE_DISABLE: ${{ vars.CABAL_CACHE_DISABLE }}
+  CABAL_CACHE_NONFATAL: ${{ vars.CABAL_CACHE_NONFATAL }}
+
+jobs:
+  build-linux:
+    name: Build linux binaries
+    runs-on: ubuntu-latest
+    env:
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      S3_HOST: ${{ secrets.S3_HOST }}
+      TARBALL_EXT: tar.xz
+      ARCH: 64
+      DEBIAN_FRONTEND: noninteractive
+      TZ: Asia/Singapore
```

Comment:
This is only relevant to S3, since there were problems with timezone parsing in cabal-cache. I had set the timezone to the bucket location initially. It doesn't really matter.